File: VB/MyLayoutView/MyLayoutHandler.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text
Imports System.Threading.Tasks
Imports DevExpress.XtraGrid.Views.Layout.Handler
Imports DevExpress.XtraGrid.Views.Layout
Imports System.Windows.Forms

Namespace MyXtraGrid
	Public Class MyLayoutViewHandler
		Inherits LayoutViewHandler
		Public Sub New(ByVal view As DevExpress.XtraGrid.Views.Layout.LayoutView)
			MyBase.New(view)

		End Sub
		Protected Overrides Sub OnKeyDown(ByVal e As KeyEventArgs)
			MyBase.OnKeyDown(e)
			If e.KeyData = Keys.Delete AndAlso View.State = LayoutViewState.Normal Then
				View.DeleteRow(View.FocusedRowHandle)
			End If
		End Sub
	End Class
End Namespace

File: CS/MyLayoutView/MyLayoutHandler.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using DevExpress.XtraGrid.Views.Layout.Handler;
using DevExpress.XtraGrid.Views.Layout;
using System.Windows.Forms;

namespace MyXtraGrid
{
    public class MyLayoutViewHandler : LayoutViewHandler
    {
        public MyLayoutViewHandler(DevExpress.XtraGrid.Views.Layout.LayoutView view)
            : base(view)
        {
            
        }
        protected override void OnKeyDown(KeyEventArgs e)
        {
            base.OnKeyDown(e);
            if (e.KeyData == Keys.Delete && View.State == LayoutViewState.Normal)
                View.DeleteRow(View.FocusedRowHandle);
        }
    }
}
